### PR TITLE
Add tests regarding inc_max

### PR DESCRIPTION
**Pull request by shiatsumat**
Add a few tests regarding inc_max.

### CODE REVIEW

File: creusot/tests/should_succeed/inc_max_repeat.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+fn inc_max_repeat(mut a: i32, mut b: i32, n: i32) {
+  let mut i: i32 = 0;
+  #[invariant(cntr_bound, i < n)]
+  #[invariant(diff_bound, (a - b).abs() >= i)]
+  while i < n {
+    let mc = take_max(&mut a, &mut b);
+    *mc += 1;
+    i += 1;
+  }
+  assert!((a - b).abs() >= i);
```
Comment (xldenis) [inline creusot/tests/should_succeed/inc_max_repeat.rs:26]: hmm for some reason this causes an error in `cargo test --test why3`

Comment (xldenis) [reply creusot/tests/should_succeed/inc_max_repeat.rs:26]: Turns out this is a case of #35, using the program function `i32::abs` in specs.

File: creusot/tests/should_succeed/inc_max_3.rs [hunk 2]
```diff
@@ -3,23 +3,30 @@
 
 extern crate creusot_contracts;
 use creusot_contracts::*;
-use std::mem::swap;
 
+#[requires(*ma <= 1000_000u32 && *mb <= 1000_000u32 && *mc <= 1000_000u32)]
+#[ensures(^ma != ^mb && ^mb != ^mc && ^mc != ^ma)]
 fn inc_max_3<'a>(ma: &'a mut u32, mb: &'a mut u32, mc: &'a mut u32) {
   if *ma < *mb {
-    swap(&mut ma, &mut mb);
```
Comment (xldenis) [inline creusot/tests/should_succeed/inc_max_3.rs]: why did you inline this? can we not define a function `swap` locally that does what these 3 lines of code do?

**Comment by xldenis** [comment]
1 comment but otherwise LGTM

**Comment by shiatsumat** [comment]
I introduced a local swap function for `inc_max_3`, which made the verification at Why3 succeed!

**Comment by shiatsumat** [comment]
Now it's ready!

**Pull request merged**